operator_csv_libs/imagerepo.py

```diff
@@ -200,6 +202,84 @@ def _get_quay_repo(self):
         r = self.image.get_image_repo().replace('quay.io/','')
         return '/'.join([r, self.image.get_image_name()])
 
+class DockerRepo:
+    
+    def __init__(self,image, docker_user=None, docker_key=None):
+        self.image = image
+
+        if docker_user:
+            self.docker_user = docker_user
+        elif 'DOCKER_USER' in os.environ:
+            # Go to environment variables if we don't have them as class variable either
+            self._docker_user = os.getenv('DOCKER_USER')
+            self.docker_user = self._docker_user
+        else:
+            self.docker_user = None
+
+        if docker_key:
+            self.docker_key = docker_key
+        elif 'DOCKER_KEY' in os.environ:
+            self._docker_key = os.getenv('DOCKER_KEY')
+            self.docker_key = self._docker_key
+        else:
+            self.docker_key = None
+
+    def get_image_digest(self):
+        return self._get_digest(manifest_list=False)
+
+    def get_manifest_list_digest(self):
+        return self._get_digest(manifest_list=True)
+
+    def get_raw_manifest_list(self):
+        if self.docker_key is not None and self.docker_user is not None:
+            skopeoCommand = "skopeo inspect --creds " + self.docker_user + ":" + self.docker_key + " --override-os linux --raw docker://" + self.image.get_image()
```

Comment:
It feels to me placing a dependency on external skopeo binary over doing this in a python native way leaves us a bit vulnerable.
I believe docker hub has a perfectly good rest api, and to simplify things there's likely exiting python libraries we can use to interface with it?